dev/NavigationView/NavigationView.cpp

```diff
@@ -1980,8 +1981,10 @@ void NavigationView::SetSelectedItemAndExpectItemInvokeWhenSelectionChangedIfNot
 
         m_indexOfLastSelectedItemInTopNav = m_topDataProvider.IndexOf(item); // for the next time we animate
     }
-
-    SelectedItem(item);
+    // Only select item when it is NOT null
+    if (item) {
```

Comment:
It may block null selections and potentially the following may be impacted:
1. customer set selectedItem to null from NavigationView API
2. click on a selected item to unselect it. so you block the fix of this: https://github.com/microsoft/microsoft-ui-xaml/issues/957#issuecomment-537265023

Reply:
Removed the null check to prevent any problems that could have been possible be caused by that.